Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -521,6 +527,8 @@ class BrowserViewController: UIViewController {
         log.debug("BVC calling super.viewDidAppear.")
         super.viewDidAppear(animated)
         log.debug("BVC done.")
+
+        delegate?.browserViewControllerDidAppear(self, profile: profile)
```

Comment:
Follow-up: check for network connectivity before trying to open the tab. You can use the various `Reachability` libraries to do this, or import `SystemConfiguration` yourself, or fetch the page contents here, or add a variant of `openURLInNewTab` that waits for a successful fetch before opening the tab and running a provided callback.